azerothcore/azerothcore-wotlk
# feat(Core/SAI): Allow forcing creatures/gameobjects's respawn timers …

…when using SMART_ACTION_FORCE_DESPAWN



## Changes Proposed:
-  Allow setting respawn timers when using SMART_ACTION_FORCE_DESPAWN (TC port)
-  

## Issues Addressed:

- Closes 

## SOURCE:


## Tests Performed:

- Tested in-game.
- 


## How to Test the Changes:

```sql
UPDATE `smart_scripts` SET `action_param2` = 35 WHERE `action_type` = 41 AND `target_type` = 10 AND `entryorguid` IN (-12168, -12173, -12164, -12166)
```

quest add 9740
go creature id 17886
Use the crystals, notice the triggers respawn in 35 seconds

```sql
-- REVERT
UPDATE `smart_scripts` SET `action_param2` = 0 WHERE `action_type` = 41 AND `target_type` = 10 AND `entryorguid` IN (-12168, -12173, -12164, -12166)
```
## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
